### PR TITLE
Feature/add saleor header for authorization

### PR DESCRIPTION
**Pull request by korycins**
I want to merge this change because it introduces a new header responsible for authorizing a user's request.

# Impact

* [ ] New migrations
* [ ] New/Updated API fields or mutations
* [ ] Deprecated API fields or mutations
* [ ] Removed API types, fields, or mutations
* [ ] Documentation needs to be updated

# Pull Request Checklist

* [ ] Privileged queries and mutations are guarded by proper permission checks
* [ ] Database queries are optimized and the number of queries is constant
* [ ] Database migration files are up to date
* [ ] The changes are tested
* [ ] GraphQL schema and type definitions are up to date
* [ ] Changes are mentioned in the changelog

### CODE REVIEW

**Comment by patrys** [comment]
Task linked: [SALEOR-2592 Add custom authorization header to saleor core](https://app.clickup.com/t/2549495/SALEOR-2592)

**Review by IKarbowiak — APPROVED** [review]

**Review by maarcingebala — APPROVED** [review]

File: saleor/core/tests/test_auth_backend_saleor_header.py [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+    staff_access_token = create_access_token(staff_user)
+    customer_access_token = create_access_token(customer_user)
+
+    request = rf.request(
+        HTTP_AUTHORIZATION_BEARER=f"{staff_access_token}",
+        HTTP_AUTHORIZATION=f"JWT {customer_access_token}",
+    )
+    backend = JSONWebTokenBackend()
+    user = backend.authenticate(request)
+    assert user == staff_user
```
Comment (patrys) [inline saleor/core/tests/test_auth_backend_saleor_header.py:30]: I would like to see a test where `authorization-bearer` is a valid Saleor token, and `authorization` is not (and is either `Basic` or `Bearer` with an unsupported token) to mirror cases where Saleor's API is sitting behind third-party authentication.

**Review by patrys — DISMISSED** [review]
Need to also whitelist the custom header in our CORS responses ("allowed headers").

File: saleor/core/tests/test_auth_backend_saleor_header.py (hunk 1 above)
Comment (korycins) [reply saleor/core/tests/test_auth_backend_saleor_header.py:44]: Added test for this.

**Comment by korycins** [comment]
> Need to also whitelist the custom header in our CORS responses ("allowed headers").

I notified the team responsible for this.

**Review by maarcingebala — APPROVED** [review]

**Pull request merged**